srtcore/core.cpp

```diff
@@ -414,10 +414,10 @@ void CUDT::setOpt(SRT_SOCKOPT optName, const void* optval, int optlen)
         if (m_bOpened)
             throw CUDTException(MJ_NOTSUP, MN_ISBOUND, 0);
 
-        if (*(int *)optval < int(CPacket::UDP_HDR_SIZE + CHandShake::m_iContentSize))
+        if (*(const int *)optval < int(CPacket::UDP_HDR_SIZE + CHandShake::m_iContentSize))
```

Comment:
This doesn't make any sense at all. Using const makes sense if it gives you a chance to be reported by the compiler when you misuse it for writing. Here it changes nothing - C-style casting can manipulate with the const modifier as it pleases.

Better idea is to introduce a local variable like
```
const int optint = *(int*)optval;
```